src/assets/wise5/authoringTool/project-asset-authoring/project-asset-authoring.component.ts

```diff
@@ -0,0 +1,292 @@
+import { Component, Inject, Optional } from '@angular/core';
+import { ConfigService } from '../../services/configService';
+import { ProjectAssetService } from '../../../../app/services/projectAssetService';
+import * as $ from 'jquery';
+import { Subscription } from 'rxjs';
+import { isImage, isVideo } from '../../common/file/file';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+
+@Component({
+  selector: 'project-asset-authoring',
+  templateUrl: './project-asset-authoring.component.html',
+  styleUrls: ['./project-asset-authoring.component.scss']
+})
+export class ProjectAssetAuthoringComponent {
+  allowedFileTypes: string[] = ['any'];
+  assetIsImage: boolean;
+  assetIsVideo: boolean;
+  assetSortBy = 'aToZ';
+  componentId = null;
+  errorFiles: any;
+  isPopup = false;
+  nodeId = null;
+  previewAssetURL: string;
+  projectId: number;
+  projectAssets: any;
+  projectAssetTotalSizeMax = 0;
+  projectAssetUsagePercentage: any = 0;
+  selectedAssetItem: any;
+  subscriptions: Subscription = new Subscription();
+  successFiles: any;
+  target = null;
+  targetObject = null;
+  totalFileSize = 0;
+  totalUnusedFilesSize = 0;
+  unusedFilesPercentage = 0;
+  uploadSuccessMessage = '';
+  uploadErrorMessage: any;
+
+  constructor(
+    private configService: ConfigService,
+    @Optional() @Inject(MAT_DIALOG_DATA) private dialogData: any,
+    @Optional() protected dialogRef: MatDialogRef<ProjectAssetAuthoringComponent>,
+    private projectAssetService: ProjectAssetService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.dialogData != null) {
+      this.setParams(this.dialogData);
+    }
+
+    this.subscriptions.add(
+      this.projectAssetService.getProjectAssets().subscribe((projectAssets) => {
+        if (projectAssets != null) {
+          this.projectAssets = projectAssets;
+          this.sortAssets(this.assetSortBy);
+          this.projectAssetTotalSizeMax = this.projectAssetService.totalSizeMax;
+        }
+      })
+    );
+
+    this.subscriptions.add(
+      this.projectAssetService.getTotalFileSize().subscribe((totalFileSize) => {
+        this.setTotalFileSize(totalFileSize);
+      })
+    );
+
+    this.subscriptions.add(
+      this.projectAssetService.getTotalUnusedFileSize().subscribe((totalUnusedFilesSize) => {
+        this.setTotalUnusedFilesSize(totalUnusedFilesSize);
+      })
+    );
+
+    if (this.projectAssetService.isProjectAssetsAvailable()) {
+      this.projectAssetService.calculateAssetUsage();
+    }
+  }
+
+  private setParams(params: any): void {
```

Comment:
Do we need these fields [isPopup/nodeId/componentId], or can they be directly obtained from this.dialogData.[isPopup/nodeId/componentId...]?

Reply:
What happens when dialogData is null?